### PR TITLE
Add XLAStatsMonitor Callback

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Add XLAStatsMonitor Callback

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/callbacks/xla_stats_monitor.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+
+if _TPU_AVAILABLE:
+    import torch_xla.core.xla_model as xm
+
+
+class XLAStatsMonitor(Callback):
+    """
+    Automatically monitors and logs XLA stats during training stage. ``XLAStatsMonitor``
+    is a callback and in order to use it you need to assign a logger in the ``Trainer``.
```
[the remark this answers is not in the document]
Comment (awaelchli) [reply pytorch_lightning/callbacks/xla_stats_monitor.py:34]: ok, nvm it is consistent with the other callback

**Review by awaelchli — APPROVED** [review]
let's also add it to callbacks.rst (the autosummary for the api overview)?

File: pytorch_lightning/callbacks/__init__.py [hunk 2]
```diff
@@ -25,13 +25,15 @@
...
 from pytorch_lightning.callbacks.timer import Timer
+from pytorch_lightning.callbacks.xla_stats_monitor import XLAStatsMonitor
 
 __all__ = [
     'BackboneFinetuning',
     'BaseFinetuning',
     'Callback',
     'EarlyStopping',
     'GPUStatsMonitor',
+    'XLAStatsMonitor',
```
[the remark this answers is not in the document]
Comment (kaushikb11) [reply pytorch_lightning/callbacks/__init__.py:36]: Good question. Thought about it, it's using the XLA library. I haven't really explored the XLA + GPUs training yet. [Ref](https://twitter.com/wightmanr/status/1384894577308471296)

Would like it to be generic for now.

File: pytorch_lightning/callbacks/xla_stats_monitor.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+    Args:
+        should_print_info: Set to ``True`` to print average peak and free memory, and epoch time
```
[the remark this answers is not in the document]
Comment (rohitgr7) [reply pytorch_lightning/callbacks/xla_stats_monitor.py:37]: how about just `verbose=True/False`?

[the remark this answers is not in the document]
Comment (Borda) [reply pytorch_lightning/callbacks/xla_stats_monitor.py:37]: Yep, sounds good, eventually it can be int as logging level

[the remark this answers is not in the document]
Comment (kaushikb11) [reply pytorch_lightning/callbacks/xla_stats_monitor.py]: Nice suggestion! Updated.

**Review by SeanNaren — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
